include/zephyr/bluetooth/services/ots.h

```diff
@@ -725,6 +725,19 @@ struct bt_ots_cb {
 	 */
 	void (*obj_name_written)(struct bt_ots *ots, struct bt_conn *conn,
 				 uint64_t id, const char *cur_name, const char *new_name);
+
+	/** @brief Object Calculate checksum callback
+	 *
+	 *  This callback is called when the OACP Calculate Checksum procedure is performed.
+	 *  Because object data is opaque to OTS service, the OTS user can only know where
+	 *  data is.
+	 *
+	 *  @param id         Object ID.
+	 *  @param offset     The first octet of the object contents need to be calculated.
+	 *  @param length     The length number of octets object name.
+	 */
+	uint32_t (*obj_cal_checksum)(uint64_t id, off_t offset, size_t len);
```

Comment:
Missing documentation of the return value. I assume it should return the checksum? 

Since the callback contains offset and len, does the functionality work such that a client can request a checksum of a subset of the object?

Reply:
1. Description of return will be added.

1. Yes, Calculate Checksum Procedure could request a subset of the object.

![image](https://user-images.githubusercontent.com/1169294/192174220-6d9a96b5-6853-4208-a688-fb531bf17344.png)